src/main/java/seedu/address/logic/commands/recipe/RecipeRemoveIngredientCommand.java

```diff
@@ -50,29 +55,32 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         assert (index.getZeroBased() < lastShownList.size());
+
         Recipe recipeToEdit = lastShownList.get(index.getZeroBased());
         UniqueIngredientList ingredients = recipeToEdit.getIngredients();
 
         try {
-            ingredients.remove(toRemove);
```

Comment:
Modifying the `UniqueIngredientList` directly violates abstraction of `Model`. The command should be getting the `Model` to modify the recipe through `EditRecipeDescriptor`. This causes other tests to fail if the test for this is run first, as it will directly modify the recipe's ingredients.